server/src/views/frame.js

```diff
@@ -115,14 +64,14 @@ class Snippet extends React.Component {
     }
 
     return <div className="snippet-container">
-      <a href={'#clip=' + encodeURIComponent(clip.id)}>
+      <a href={'?clip=' + encodeURIComponent(clip.id)}>
         { node }
       </a>
       <p>
         <img ref="commentBubble" className="comment-bubble"
           src={ closed ? this.props.staticLink("img/comment-bubble.png") : this.props.staticLink("img/comment-bubble-open.png") }
           onClick={ this.onClickComment.bind(this) } />
-        <a href={'#clip=' + encodeURIComponent(clip.id)}>
+        <a href={'?clip=' + encodeURIComponent(clip.id)}>
```

Comment:
Yes. We can't use the hash for the clip id, because react requires rendering the same thing on the server and the client, and the hash doesn't get sent to the server.

Before, react-router was handling partial page refreshes and pushState, but we have to do it ourselves if we want that now.